commercial/app/controllers/commercial/LiveEventsController.scala

```diff
@@ -0,0 +1,38 @@
+package controllers.commercial
+
+import common.{ExecutionContexts, Logging}
+import model.NoCache
+import model.commercial.events.LiveEventAgent
+import performance.MemcachedAction
+import play.api.mvc._
+
+import scala.concurrent.Future
+
+object LiveEventsController
+  extends Controller
+  with ExecutionContexts
+  with Logging
+  with implicits.Collections
+  with implicits.Requests {
+
+  def renderEvent = MemcachedAction { implicit request =>
+    specificId map { eventId =>
+        Future {
+          val clickMacro = request.getParameter("clickMacro")
+          val omnitureId = request.getParameter("omnitureId")
+          val selectedLiveEvents = LiveEventAgent.specificLiveEvents(specificIds)
```

Comment:
It looks like you only need to find a single event so could use `eventId` instead of `specificIds`.